### ISSUE TITLE
Provide lots of use case examples, explain design decision, etc., in introduction

### ISSUE THREAD

**Issue by slightlyoff** [WICG/feature-policy#29]
It's fine for this effort to have a spec, but design shouldn't start there. It'd be better to outline the needs and proposed solutions in the form of a design doc & example code. E.g., see the Intersection Observer explainer: https://github.com/WICG/IntersectionObserver/blob/gh-pages/explainer.md

**Comment by igrigorik** [comment]
Yep, we definitely need to expand our current intro:
 - explain the link/merge with permission delegation /cc @raymeskhoury
 - explain the link with feature policy (enable policy)
 - more hands-on examples..

That said, a meta question.. @slightlyoff: why do we advocate for standalone 'explainers' instead of keeping that same content in the introduction section of the spec? Personally I'd prefer to avoid duplication and keep everything everything in the same document for versioning and consistency.

**Comment by slightlyoff** [comment]
Explainers do a few positive things to the design process:

 - They force you to write example code and not IDL. Us "browser people" tend to write out formal language and IDL as though it somehow makes sense as a way to debate the nooks and crannies of the design space. Particularly early in the design, this is unnecessary and unwelcome over-specificity.
 - Because Explainers tend to be mostly problem descriptions and example code, they allow potential users to evaluate the proposed design more directly and provide quality feedback. Put another way, they keep us honest. Explainers don't allow designs to hide behind spec jargon.
 - They allow you to delay the process of writing formal spec text until something is _actually_ designed. This allows you to keep your agility early on.

Basically my comment is that the spec is premature. We should do design and forge agreement on that _before_ we move to formal spec text.

**Comment by igrigorik** [comment]
Had a quick offline chat with @slightlyoff today.. Long story short: +1 to all of the above as best practice for early stages in exploring the problem space. For FP in particular, we're already well on the way ([blink itent to implement](https://groups.google.com/a/chromium.org/forum/#!topic/blink-dev/rBYR3Xd6Skw)) and at a stage where nailing down the edge cases in a formal spec is the right place to iterate. That said, we did agree that we should add "boatloads of examples" in introduction section of the spec to clarify intent, major design decisions, and features. 

With that in mind, leaving this open until we satisfy the "boatloads" requirement.. :-)

**Comment by intchloe** [comment]
I think the draft fails to describe the syntax. Do examples really need the HTTP-JFV syntax?

**Comment by igrigorik** [comment]
WIP explainer: https://docs.google.com/document/d/1k0Ua-ZWlM_PsFCFdLMa8kaVTo32PeNZ4G7FFHqpFx4E/edit. I'll also link it from the README.
